catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java

```diff
@@ -177,14 +179,12 @@ public final Attribute getAttribute(String name) {
     if (Metacard.ID.equals(name)) {
       return this.id;
     }
-    for (Attribute attribute : attributes) {
-      if (attribute == null || StringUtils.isEmpty(attribute.getName())) {
-        continue;
-      } else if (name.equals(attribute.getName())) {
-        return attribute;
-      }
-    }
-    return null;
+    return attributes
+        .stream()
+        .filter(attr -> !StringUtils.isEmpty(attr.getName()))
+        .filter(attr -> name.equals(attr.getName()))
```

Comment:
✏️ Instead of filtering twice against the attribute's name, could filter once: `.filter(attr -> name.equals(attr.getName()))` will test accurately against `null`.